File: src/1773-Count-Items-Matching-a-Rule/count_items_matching_a_rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
  let ans = 0
  let index

  if (ruleKey === "type") {
    index = 0
  } else if (ruleKey === "color") {
    index = 1
  } else if (ruleKey === "name") {
    index = 2
  }

  for (let item of items) {
    if (item[index] === ruleValue) {
      ans += 1
    }
  }

  return ans
};

const items = [["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]]
const ruleKey = "color"
const ruleValue = "silver"

console.log(solution.countMatches(items, ruleKey, ruleValue))

File: src/1773-Count-Items-Matching-a-Rule/count_items_matching_a_rule.cpp
#include <iostream>
#include <vector>
#include <string>

class Solution {
public:
    int countMatches(const std::vector<std::vector<std::string>>& items, const std::string& ruleKey, const std::string& ruleValue) {
        int ans = 0;
        int index = 0;

        if (ruleKey == "type") {
            index = 0;
        } else if (ruleKey == "color") {
            index = 1;
        } else if (ruleKey == "name") {
            index = 2;
        }

        for (const auto& item : items) {
            if (item[index] == ruleValue) {
                ans += 1;
            }
        }

        return ans;
    }
};

int main() {
    std::vector<std::vector<std::string>> items = {{"phone", "blue", "pixel"}, {"computer", "silver", "lenovo"}, {"phone", "gold", "iphone"}};
    std::string ruleKey = "color";
    std::string ruleValue = "silver";
    Solution solution;
    
    std::cout << solution.countMatches(items, ruleKey, ruleValue);

    return 0;
}
